tooling/nargo_cli/src/cli/expand_cmd.rs

```diff
@@ -0,0 +1,224 @@
+    let parsed_files = parse_all(&workspace_file_manager);
+
+    for package in &workspace {
+        expand_package(&workspace_file_manager, &parsed_files, package, &args.compile_options)?;
+    }
+
+    Ok(())
+}
+
+fn expand_package(
+    file_manager: &FileManager,
+    parsed_files: &ParsedFiles,
+    package: &Package,
+    compile_options: &CompileOptions,
+) -> Result<(), CompileError> {
+    let code = get_expanded_package(file_manager, parsed_files, package, compile_options)?;
+    println!("{code}");
+    Ok(())
+}
+
+fn get_expanded_package(
+    file_manager: &FileManager,
+    parsed_files: &ParsedFiles,
+    package: &Package,
+    compile_options: &CompileOptions,
+) -> Result<String, CompileError> {
+    let (mut context, crate_id) = prepare_package(file_manager, parsed_files, package);
+
+    // Even though this isn't LSP, we need to active this to be able to go from a ModuleDefId to its parent module
+    context.activate_lsp_mode();
+
+    check_crate_and_report_errors(&mut context, crate_id, compile_options)?;
+
+    let root_module_id = context.def_maps[&crate_id].root();
+    let module_id = ModuleId { krate: crate_id, local_id: root_module_id };
+
+    let mut builder = ItemBuilder::new(crate_id, &context.def_interner, &context.def_maps);
+    let item = builder.build_module(module_id);
+
+    let mut string = String::new();
+    let mut printer =
+        ItemPrinter::new(crate_id, &context.def_interner, &context.def_maps, &mut string);
+    printer.show_item(item);
+
+    let (parsed_module, errors) = parse_program_with_dummy_file(&string);
+    if errors.is_empty() {
+        let config = nargo_fmt::Config {
+            reorder_imports: true,
+            imports_granularity: ImportsGranularity::Crate,
+            ..Default::default()
+        };
+        Ok(nargo_fmt::format(&string, parsed_module, &config))
+    } else {
+        string.push_str("\n\n// Warning: the generated code has syntax errors");
+        Ok(string)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use nargo::{insert_all_files_for_workspace_into_file_manager, parse_all};
+    use nargo_toml::PackageSelection;
+    use noirc_frontend::elaborator::UnstableFeature;
+
+    use crate::cli::read_workspace;
+
+    use super::{ExpandCommand, get_expanded_package};
+
+    #[test]
+    fn nargo_expand_smoke_test() {
+        let root_path = std::env::current_dir()
+            .unwrap()
+            .join("test_programs")
+            .join("expand")
+            .canonicalize()
+            .expect("Could not resolve root path");
+
+        let mut command = ExpandCommand::default();
+        command.compile_options.unstable_features.push(UnstableFeature::Enums);
+        let workspace = read_workspace(&root_path, PackageSelection::All).unwrap();
+
+        let mut workspace_file_manager = workspace.new_file_manager();
+        insert_all_files_for_workspace_into_file_manager(&workspace, &mut workspace_file_manager);
```

Comment:
Why add all of the `test_programs/expand` packages in a smoke test? Does this test overlap with the `nargo_cli` tests?

Reply:
Oh, there's another `test_programs/expand` directory inside `nargo_cli`. But maybe this smoke tests shouldn't be needed anymore now that we test almost every test program. What do you think?

Reply:
I decided to remove it as most programs in `test_programs` are small and it should be easy to debug them because `nargo expand` works on the main crate, so if there's not much code there there won't be a lot of logic happening there, even if that crate has several dependencies.